fe/fe-core/src/main/java/org/apache/doris/datasource/ExternalRowCountCache.java

```diff
@@ -86,56 +86,50 @@ protected Optional<Long> doLoad(RowCountKey rowCountKey) {
                 TableIf table = StatisticsUtil.findTable(rowCountKey.catalogId, rowCountKey.dbId, rowCountKey.tableId);
                 return Optional.of(table.fetchRowCount());
             } catch (Exception e) {
-                LOG.warn("Failed to get table with catalogId {}, dbId {}, tableId {}", rowCountKey.catalogId,
-                        rowCountKey.dbId, rowCountKey.tableId);
-                return Optional.empty();
+                LOG.warn("Failed to get table row count with catalogId {}, dbId {}, tableId {}. Reason {}",
+                        rowCountKey.catalogId, rowCountKey.dbId, rowCountKey.tableId, e.getMessage());
+                LOG.debug(e);
+                return null;
```

Comment:
why return `null` instead of `Optional.empty()`. null will raise npe when do `orElse()` in `getCachedRowCount`?

Reply:
Return Optional.empty() will cache this empty value in memory, so we can't try to load the row count until the cache expire. Null may raise NPE, but that is expected, we catch that NPE and return a default value -1 without keep the default value or Optinal.empty() in cache, so we can trigger the load function to fetch row count again next time when exception happened.

Reply:
To add this explanation to a comment?

Reply:
added